### PR TITLE
Use VersionNumber for libllvm_version

### PR DESCRIPTION
**Pull request by musm**
To me it looks like every usage of `libllvm_version` is better of if it was an actual `VersionNumber`

Same with packages that use this which also just end up wrapping the string in a `VersionNumber`. 

This makes things more convenient.

### CODE REVIEW

**Comment by musm** [comment]
bump

**Comment by musm** [comment]
Also should `libllvm_version` be  uppercase ?

File: base/Makefile [hunk 1]
```diff
@@ -55,7 +55,7 @@ else
 endif
 	@echo "const libfftw_name = \"$(LIBFFTWNAME)\"" >> $@
 	@echo "const libfftwf_name = \"$(LIBFFTWFNAME)\"" >> $@
-	@echo "const libllvm_version = \"$$($(LLVM_CONFIG_HOST) --version)\"" >> $@
+	@echo "const libllvm_version_string = \"$$($(LLVM_CONFIG_HOST) --version)\"" >> $@
```
Comment (tkelman) [inline base/Makefile]: if possible to avoid occupying two names in base, would be better to do the VersionNumber conversion here

Comment (musm) [inline base/Makefile]: Yeah I agree, but there isn't a way to do this without making unnecessary changes/shuffling of files.   `VERSION_STRING` follows the same strategy as in this PR.

Any opinions on the uppercasing (in regards to consistency) ?

**Comment by musm** [comment]
On second thought I think it's probably better to not touch the the uppercasing (at least for this PR)

**Comment by musm** [comment]
bump

**Review by StefanKarpinski — APPROVED** [review]

**Comment by StefanKarpinski** [comment]
Meant to squash, but clicked twice by accident.

**Comment by musm** [comment]
thanks!  Both comments are valid so I think it's fine
